### ISSUE TITLE
Training Demo

### ISSUE THREAD

**Issue by copperbeardy** [Lombiq/Orchard-Training-Demo-Module#23]
I have been looking at the training

 https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/orchard-core/StartLearningHere.md

Interesting read some may find it a bit cumbersome to go through code files to find information.

I wonder if you have thought of using Youtube/Twitch to do live online stream demos, where you can interact with the public directly, easily and pretty much get instant feedback.

I know you have the Meeting you share and I think people can join in but some might not know that as it is not hugely visible. 

anyway just a thought as some prefer audio & video demos to read

**Comment by Piedone** [comment]
Thanks for the suggestion!

The Training Demo is for people who like to learn from code and who want to have a thoroughly annotated code reference on how to do everything in Orchard. It's certainly not for everybody.

Just as we've done for Orchard 1.x we're also working on a video tutorial series for Orchard Core, which will supplement this module with a different format.

Having live coding sessions is an interesting idea. However, to do it right the one streaming would need to a) explain what they're doing and b) work on something that can actually be public. Both of these make these not so simple, and our resources being finite too I don't think we'd be able to do it in the short term. But we'll certainly look into it and see what we can do!

**Comment by copperbeardy** [comment]
yeah, I can understand that, looking forward to the video series. I stream myself and have been trying to develop a project using Orchard core. But I am still trying to get my head around a lot of it  :)

**Comment by Piedone** [comment]
FYI we're having an internal discussion on potentially streaming something in the near future. The bottleneck here is that right now we're almost entirely working on client projects and naturally, we can't show the code of those.

**Comment by Piedone** [comment]
Would you have some suggestions for such live demos or coding sessions? Like any particular topic you're interested? BTW we'd use YouTube as we already have all our videos there.

**Comment by copperbeardy** [comment]
yeah, I have seen the youtube Vod, Podcasts and am in the Gitter channel as well :)  

Since people can work with Orchard core in the Code and within the CMS GUI. 

demoing the flexibility of the platform showing how something is done in both codes and in the CMS GUI and how those objects can be interacted with by the code and CMS GUI., for example, user-created items that are created through the developer's code that can then be viewed in CMS GUI. the various ways to work with persisted data.

creating something in code or in the backend and how those items can leverage the Workflows, to make richer applications

the Various methods you provide to secure applications.

what you would consider best practices for how people code against packages Orchard core.

hope thats is helpful

**Comment by Piedone** [comment]
I see, thanks for elaborating!

This is something that we'd cover during our tutorial series. In the short term I could do a walkthrough of the Publish Later module's code (https://www.youtube.com/watch?v=E7UH8R14EUA&lc=z23ihl4jdpmgepk5b04t1aokgdt0wu52ptrx5xoc1catrk0h00410&feature=em-comments). This would touch on several points you mention.

**Comment by copperbeardy** [comment]
cool will have a look through it

**Comment by Piedone** [comment]
Not sure when we'll have a coding session like this but will keep in mind. I don't think there's anything to do here though.
